### DISCUSSION TITLE
Calculate wake while skipping deflection calculation suggestion

### DISCUSSION THREAD

**Discussion by Maartenvbeek** [NatLabRockies/floris#270] [Ideas]
Hi,

Currently I have to calculate the wake effects many times for different time stamps. In order to speed up the calculation, I dived a bit more into the FLORIS package and the possibilities. I noticed that when the function `calculate_wake()` is called, there is no option to avoid the wake deflection calculation, happening in the flow_field.py simulation file lines 639-641: 

```
# get the wake deflection field
deflection = self._compute_turbine_wake_deflection(
    rotated_x, rotated_y, rotated_z, turbine, coord, self
)
```

Would it be possible to add a boolean to `calculate_wake()` called e.g. "no_wake_deflection" making the above code something like:

```
# get the wake deflection field
if not no_wake_deflection:
    deflection = self._compute_turbine_wake_deflection(
              rotated_x, rotated_y, rotated_z, turbine, coord, self
          )
else:
    deflection = np.zeros()
```

I am not sure what size the zeros array should be, but it is probably doable to make it the right size based on other arrays already known.

Another suggestion I just thought off: make it possible to select in the .json file "None" as option for the "deflection_model", similar to what is possible for the "turbulence_model".

It would save me quite some time when running part of the script where I am not interested in the wake deflection.

Please let me know what you think of my suggestion.

Kind regards,

Maarten

**Comment by Bartdoekemeijer** [comment]
Hi Maarten,

I really like this idea. Our apologies for the delayed response. I've opened Issue #394 to get this feature implemented in FLORIS v3.
